### ISSUE TITLE
Atomic Update command and specail symbols (quotes)

### ISSUE THREAD

**Issue by mr-KVA** [SolrNet/SolrNet#445]
Atomic Update method doesn't handle values with spec symbols, for instance - quotes (").

https://github.com/SolrNet/SolrNet/blob/b95e39b0c521fa92088712351f7331ea648b4fd9/SolrNet/Commands/AtomicUpdateCommand.cs#L76-L87

Is there any reason why 3rd party lib (like Json.NET) wasn't used in json serialization?

**Comment by ParkerBeck** [comment]
I've ran into this as well. I fixed this by adding the Json.NET package to SolrNet project and serializing `uniqueKey`, `id`, `updateSpec.Field`, and `updateSpec.Value` through `JsonConvert.SerializeObject()`. This will notably also fail with other escape characters and can lead to injection attacks. Any data with `\n`, `\t`, and `\r` will also be treated incorrectly.

This DOES introduce the Json.NET dependency into SolrNet.Core, but it's already using this package in SolrNet.Cloud so it seems acceptable.

Looking through the rest of the code base, all other commands are serialized into xml automatically which is why this isn't a problem with anything else. Maybe the correct solution would be to reach parity by switching `GetAtomicUpdateJson()` to `GetAtomicUpdateXml()`?
